components/suggest/src/db.rs

```diff
@@ -945,6 +945,18 @@ impl<'a> SuggestDao<'a> {
     /// Deletes all suggestions associated with a Remote Settings record from
     /// the database.
     pub fn drop_suggestions(&mut self, record_id: &SuggestRecordId) -> Result<()> {
+        self.conn.execute_cached(
```

Comment:
This seems totally reasonable to me, and SQLite could probably optimize this query plan much better than relying on `ON DELETE CASCADE` (which is implemented using row-level triggers under the hood).

Reply:
> SQLite could probably optimize these query plans much better than relying on ON DELETE CASCADE
Agreed and the benchmarks seem to verify this.